onnx/shape_inference/implementation.h

```diff
@@ -17,23 +17,32 @@ struct GraphInferenceContext {
       const std::unordered_map<std::string, TypeProto*>&
           outer_scope_value_types_by_name_in,
       const std::unordered_map<std::string, int> opset_imports_in,
+      SymbolTable& symbolTable_in,
       const ISchemaRegistry* schema_registry_in = OpSchemaRegistry::Instance())
       : outer_scope_value_types_by_name{&outer_scope_value_types_by_name_in},
         opset_imports{opset_imports_in},
-        schema_registry{schema_registry_in} {}
+        schema_registry{schema_registry_in},
+        symbolTable{symbolTable_in} {}
 
 
   const std::unordered_map<std::string, TypeProto*>*
       outer_scope_value_types_by_name;
   const std::unordered_map<std::string, int> opset_imports;
   const ISchemaRegistry* schema_registry;
+  SymbolTable& symbolTable;
 
+  SymbolTable* mutableSymbolTable() {
+    return &symbolTable;
+  }
 };
 
 class GraphInferencerImpl : public GraphInferencer {
  public:
   GraphInferencerImpl(GraphProto& g, const GraphInferenceContext& context)
-      : g_{&g}, context_{&context} {}
+      : g_{&g}, context_{&context} {
+        auto* contextForSymbol = const_cast<GraphInferenceContext*>(context_);
+        symbolTable_ = contextForSymbol->mutableSymbolTable();
```

Comment:
how you are making the symbolTable available to GraphInferenceImpl needs to be simplified

context_ is private var for this class... can you not simply add a public mtd to fetch symbol table from context? So you that can get rid of symbolTable_ from here